### PR TITLE
Fix Native OIDC for Element Desktop

### PR DESCRIPTION
**Pull request by t3chguy**
Requires https://github.com/matrix-org/matrix-js-sdk/pull/4068

---
Here's what your changelog entry will look like:

## ✨ Features
 * Fix Native OIDC for Element Desktop ([\#12253](https://github.com/matrix-org/matrix-react-sdk/pull/12253)).

### CODE REVIEW

File: src/BasePlatform.ts [hunk 1]
```diff
@@ -478,4 +478,11 @@ export default abstract class BasePlatform {
             policyUri: config.privacy_policy_url,
         };
     }
+
+    /**
+     * Suffix to append to the `state` parameter of OIDC /auth calls, will be round-tripped to the callback URI
+     */
+    public getOidcClientState(): string {
+        return "";
```
Comment (florianduros) [inline src/BasePlatform.ts:486]: Is this empty string desired?

Comment (t3chguy) [reply src/BasePlatform.ts:486]: Yeah, only the ElectronPlatform actually needs any data roundtripped like this

**Comment by t3chguy** [comment]
It doesn't work, it never did. The flow is impossible to complete.

File: src/BasePlatform.ts [hunk 2]
```diff
@@ -478,4 +482,12 @@ export default abstract class BasePlatform {
             policyUri: config.privacy_policy_url,
         };
     }
+
+    /**
+     * Suffix to append to the `state` parameter of OIDC /auth calls, will be round-tripped to the callback URI.
```
Comment (richvdh) [inline src/BasePlatform.ts:487]: comma splice
```suggestion
     * Suffix to append to the `state` parameter of OIDC /auth calls. Will be round-tripped to the callback URI.
```

File: src/BasePlatform.ts [hunk 3]
```diff
@@ -478,4 +482,12 @@ export default abstract class BasePlatform {
...
+     * Currently only required for ElectronPlatform for passing element-desktop-ssoid.
```
Comment (richvdh) [inline src/BasePlatform.ts:488]: is there an associated PR which will actually populate this for `ElectronPlatform` ?

**Review by richvdh — APPROVED** [review]
lgtm, though it is considerably terrifying that we seem have no tests for this?

**Pull request merged**